lib/rouge/lexer.rb

```diff
@@ -405,15 +413,18 @@ def reset!
 
     # Given a string, yield [token, chunk] pairs.  If no block is given,
     # an enumerator is returned.
-    #
-    # @option opts :continue
-    #   Continue the lex from the previous state (i.e. don't call #reset!)
-    def lex(string, opts={}, &b)
-      return enum_for(:lex, string, opts) unless block_given?
+    def lex(string, &b)
```

Comment:
Would it be better to keep the current `lex` method but deprecate `opts`? This will be a breaking change, right? Or perhaps it would be a good idea to make the next version of Rouge 4.0.0 given the amount of time since the last gem release?

Reply:
I will provide a deprecation, and default it to `nil` to avoid allocation.